### PR TITLE
Enable WC Customizer panels in customize.php routes also for block themes

### PR DESCRIPTION
**Pull request by Aljullu**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

In #34022 we removed WooCommerce panels from the Customizer when the store has a block theme enabled. However, after some discussion (see https://github.com/woocommerce/woocommerce/issues/34430 and p1661253811959289-slack-C02FL3X7KR6) we decided to still register those panels when the Customizer route is accessed directly.

This way:

1. WooCommerce panels are present if there is another extension that registers panels in the Customizer.
2. Block theme users still have a way to access the customizer directly from the URL (currently, the work-around was to switch to a classic theme to make any modifications).

### How to test the changes in this Pull Request:

**With a classic theme (ie: Storefront)**
1. Verify links to the Customizer are shown in the admin top bar and under Appearance (in the admin sidebar).
2. Verify you can access the Customizer normally and modify settings under the WooCommerce panel.
3. Verify once you save the settings and reload the page, they are persisted.

**With a block theme (ie: Twenty Twenty Two)**
1. Verify there are no links to the Customizer anywhere: not in the admin top bar neither under Appearance (in the admin sidebar).
2. Verify you can access the Customizer if you write (`/wp-admin/customize.php` in the URL bar of your browser).
3. Verify you can modify settings under the WooCommerce panel.
3. Verify once you save the settings and reload the page, they are persisted.

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [ ] Have you successfully run tests with your changes locally?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm changelog add --filter=<project>`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Review by albarin — APPROVED** [review]
👌

File: plugins/woocommerce/includes/customizer/class-wc-shop-customizer.php [hunk 1]
```diff
@@ -885,6 +885,11 @@ public function has_terms_and_conditions_page_id() {
 	}
 }
 
-if ( ! wc_current_theme_is_fse_theme() ) {
+global $pagenow;
+if (
+	$pagenow === 'customize.php' ||
+	isset( $_REQUEST['customize_theme'] ) || // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+	! wc_current_theme_is_fse_theme()
```
Comment (nerrad) [inline plugins/woocommerce/includes/customizer/class-wc-shop-customizer.php:892]: This condition is probably not needed?

Comment (nerrad) [reply plugins/woocommerce/includes/customizer/class-wc-shop-customizer.php:892]: Oh NVM, this is needed to make sure with classic themes registration happens early enough to trigger the menu item showing.

**Comment by nerrad** [comment]
I know it would require a release exception to the code freeze, but I wonder if we should target WooCommerce 6.9 for this (cc @ObliviousHarmony). Some rationale:

- We may be jumping the gun too early in _[completely removing these settings](https://github.com/woocommerce/woocommerce/pull/34022)_ when a block theme is active.
- This PR still accomplishes the intent of #34022 but keeps the Woo Customizer settings easily accessible which could be important for supporting sites that still need to access them when they have an active block theme.
- The code change is minimal and easy to test.

If agreed, I'll follow the existing process for seeking an exception.

**Comment by Aljullu** [comment]
> I wonder if we should target WooCommerce 6.9 for this

+1 from my side

**Comment by nerrad** [comment]
Let's just go ahead and do a code freeze exception for this PR. Without it, there's no direct way for folks to still access the customizer settings directly when a block theme is active.

**Comment by nerrad** [comment]
@konamiman I see this was assigned to the WC 7.0.0 milestone when there was a code freeze exception requested to get this into WC 6.9.0. Does that mean the exception was denied?

**Comment by Konamiman** [comment]
@nerrad The milestone is assigned automatically on pull request merge, the chosen value is the version immediately after the one that is currently code-freezed. For granted exceptions the milestone needs to be changed manually (but I'm not sure who's responsible for doing that, I guess the person who requested the exception maybe?)

**Comment by Aljullu** [comment]
I just updated the milestone. :+1:

File: plugins/woocommerce/includes/customizer/class-wc-shop-customizer.php (hunk 1 above)
Comment (dinhtungdu) [reply plugins/woocommerce/includes/customizer/class-wc-shop-customizer.php:892]: I have the same question while working on https://github.com/woocommerce/woocommerce/issues/58364 (https://github.com/woocommerce/woocommerce/pull/58640). 

> this is needed to make sure with classic themes registration happens early enough to trigger the menu item showing.

@nerrad if you can recall (it's almost 3 years), what do you mean by the "menu item showing"? It seems to me that we only need the customizer class inside `customize.php` route.

Comment (nerrad) [reply plugins/woocommerce/includes/customizer/class-wc-shop-customizer.php:892]: I _think_ I'm referring to the "Customize" menu item in the sidebar (in appearance menu) or the admin bar.

Comment (dinhtungdu) [reply plugins/woocommerce/includes/customizer/class-wc-shop-customizer.php:892]: @nerrad I see, but isn't that menu controlled by WordPress, not WooCommerce, right?

Comment (nerrad) [reply plugins/woocommerce/includes/customizer/class-wc-shop-customizer.php:892]: It's registered conditionally:

https://github.com/WordPress/wordpress-develop/blob/7d5fb7b4a98a4ac02ef0d7b354f11b1396363d51/src/wp-admin/menu.php#L227-L229

Comment (dinhtungdu) [reply plugins/woocommerce/includes/customizer/class-wc-shop-customizer.php:892]: Yes, but I still think that we don't need `! wc_current_theme_is_fse_theme()` to ensure WooCommerce Customizer works as expected. But we should still check for the block theme to only initialize the class for classic themes. So yeah let's keep that.

**Pull request merged**